arraycontext/pytest.py

```diff
@@ -59,10 +59,12 @@ def pytest_generate_tests_for_pyopencl_array_context(metafunc):
     from pyopencl.tools import _ContextFactory
 
     class ArrayContextFactory(_ContextFactory):
-        def __call__(self):
+        def __call__(self, force_device_scalars=False):
```

Comment:
How about PyOpenCLArrayContext always yields 2 variants (like it does with the multiple CL-contexts) one with `force_device_scalars=True` and vice-versa.

Reply:
Not a big fan of that at the moment because it's going to force everyone using `pytest_generate_tests_for_pyopencl_array_context` to handle both cases (and break most of the downstream CIs). It should be something to opt-in for.

Maybe something closer to what you did in #14 to allow multiple array contexts?

Reply:
Gave that a go in #44. I can squash that into this or just merge it after, since this is probably more urgent for the `to_numpy` scalar fix.

Reply:
Agree with @alexfikl here. If we change the behavior of this, it'll break tons of downstream code without warning. The whole point of the flag is to allow the change to be gradual and opt-in.

Also, thanks for taking on #44, too. Will review that next, but given the below, I suspect I'd prefer them going in via the same PR.